src/rgw/rgw_crypt_sanitize.h

```diff
@@ -0,0 +1,58 @@
+// -*- mode:C; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef RGW_RGW_CRYPT_SANITIZE_H_
+#define RGW_RGW_CRYPT_SANITIZE_H_
+
+#include "rgw_common.h"
+
+namespace rgw {
+namespace crypt_sanitize {
+
+
+struct env {
+  boost::string_ref name;
+  boost::string_ref value;
+
+  env(boost::string_ref name, boost::string_ref value)
+  : name(name), value(value) {}
+};
+
+struct x_meta_map {
+  boost::string_ref name;
+  boost::string_ref value;
+  x_meta_map(boost::string_ref name, boost::string_ref value)
+  : name(name), value(value) {}
+};
+
+struct s3_policy {
+  boost::string_ref name;
+  boost::string_ref value;
+  s3_policy(boost::string_ref name, boost::string_ref value)
+  : name(name), value(value) {}
+};
+
+struct auth {
+  req_state* const s;
+  boost::string_ref value;
+  auth(req_state* const s, boost::string_ref value)
+  : s(s), value(value) {}
+};
+
+struct log_content {
+  const char* buf;
+  log_content(const char* buf)
+  : buf(buf) {}
+};
+
+}
+}
+
+namespace std {
+std::ostream& operator<<(std::ostream& out, const rgw::crypt_sanitize::env& e);
```

Comment:
these `operator<<` overloads belong with their types in namespace `rgw::crypt_sanitize`

re: `namespace std` in the c++ standard:
> 17.6.4.2.1  Namespace std [namespace.std]
> The behavior of a C++ program is undefined if it adds declarations or definitions to namespace std or to a namespace within namespace std unless otherwise specified. A program may add a template specialization for any standard library template to namespace std only if [...]